src/main/java/cdw/springtraining/gatekeeper/service/AdminService.java

```diff
@@ -0,0 +1,272 @@
+        return residentList;
+    }
+
+    public ResidentObject createNewResident(CreateResident request) throws Exception {
+
+        if (residentRepository.existsByAadhar(request.getAadhar()))
+            throw new UserAlreadyExistsException("You have already registered");
+
+        Resident resident = new Resident(request.getResidentName(), request.getAadhar(), request.getResidenceId(), request.getPhoneNumber(), true);
+        residentRepository.save(resident);
+
+        Resident response = residentRepository.findByAadhar(request.getAadhar());
+        ResidentObject residentObject = new ResidentObject();
+        residentObject.setId(response.getId());
+        residentObject.setResidentName(response.getResidentName());
+        residentObject.setAadhar(response.getAadhar());
+        residentObject.setPhoneNumber(response.getPhoneNumber());
+        residentObject.setResidenceId(response.getResidenceNumber());
+        return residentObject;
+
+    }
+
+
+    public boolean deleteResident(Integer residentId) throws Exception {
+        Resident resident = residentRepository.findById(residentId).orElseThrow(() -> new UserNotFoundException("Resident not found"));
+        resident.setActive(false);
+        residentRepository.save(resident);
+        return true;
+    }
+
+    public ResidentObject updateResident(Integer residentId, UpdateResident updateResident) throws Exception {
+        Optional<Resident> residentOptional = residentRepository.findById(residentId);
+        if (!residentOptional.get().isActive()) throw new UserHasBeenRemovedException("The resident had been deleted");
+        if (residentOptional.isPresent()) {
+            Resident resident = residentOptional.get();
+
+            if (updateResident.getResidentName() != null) {
+                resident.setResidentName(updateResident.getResidentName());
+            }
+            if (updateResident.getAadhar() != null) {
+                resident.setAadhar(updateResident.getAadhar());
+            }
+            if (updateResident.getPhoneNumber() != null) {
+                resident.setPhoneNumber(updateResident.getPhoneNumber());
+            }
+            residentRepository.save(resident);
+
+            ResidentObject residentObject = new ResidentObject();
+            residentObject.setId(resident.getId());
+            residentObject.setResidentName(resident.getResidentName());
+            residentObject.setAadhar(resident.getAadhar());
+            residentObject.setPhoneNumber(resident.getPhoneNumber());
+            residentObject.setResidenceId(resident.getResidenceNumber());
+
+            return residentObject;
+
+        } else throw new UserNotFoundException("Resident not found");
+
+    }
+
+    public ResidentObject getResidentById(Integer residentId) throws Exception {
+        Resident resident = residentRepository.findById(residentId).orElseThrow(() -> new UserNotFoundException("Resident Not found"));
```

Comment:
use constants file for strings for the whole project level

Reply:
Added strings to constants